### PR TITLE
Notifications sockets

### PR DESCRIPTION
**Pull request by DevDelvo**
- [x] - need API routes to get notifications on the front-end
- [x] - whenever someone invests in your project, broadcast an event using sockets, and on the front-end, listen for that event to show that real-time notification

### CODE REVIEW

**Comment by DevDelvo** [comment]
I'm going to improve the look and functionality when I work on it for Issue #90 FS: Improve the look and feel of the dropdown notifications

File: client/src/components/NavBar.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ import MenuIcon from '@material-ui/icons/Menu';
 
 import SvgProductLaunchLogo from './ProductLaunchLogo';
 import NotificationDropdown from './NotificationDropdown';
-import { messages } from '../dummyData/dropDownItems'; //standin data
+import { getNotifications } from '../api/notifications';
```
Comment (evankp) [inline client/src/components/NavBar.js]: @tsyaeger changed the routes to absolute, so you could do something like api/notifications, and similar for above starts from src on frontend

File: server/sockets/events/Investments.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+exports.receiveInvestment = (io, socket) => ({ id, name, projectName }) => {
+  socket.join(id);
+  console.log(`${name} invested in your project, ${projectName}!`);
```
Comment (evankp) [inline server/sockets/events/Investments.js]: You probably don't need this console log here

**Review by evankp — APPROVED** [review]

File: client/src/components/NavBar.js (hunk 1 above)
Comment (DevDelvo) [reply client/src/components/NavBar.js]: Okay I shall change it. Not used to that absolute pathing yet

File: server/sockets/events/Investments.js (hunk 2 above)
Comment (DevDelvo) [reply server/sockets/events/Investments.js]: I will keep it for debugging purposes though, similar to how we have it console log when it sockets are connected and disconnected

File: client/src/api/notifications.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import httpClient from './httpClient';
+
+export const getNotifications = async (userId) => {
```
Comment (tsyaeger) [inline client/src/api/notifications.js]: is the injection still not working for you? It works for me. Probably something we can sort out to cut down on duplicate code.

File: client/src/components/NavBar.js (hunk 1 above)
Comment (tsyaeger) [inline client/src/components/NavBar.js]: could use absolute paths (basing this on Shums' feedback for me)

File: client/src/components/NavBar.js [hunk 4]
```diff
@@ -223,11 +246,11 @@ const NavBar = (props) => {
...
           </Link>
         </div>
-        {props.notificationCount > 0 &&
+        {props.notifications && props.notifications.length > 0 &&
           <div>
             <NotificationDropdown
-              alerts={props.notificationCount}
-              messages={messages}
+              alerts={props.notifications.length}
+              notifications={props.notifications}
```
Comment (tsyaeger) [inline client/src/components/NavBar.js]: this might be named investments to be more precise

File: client/src/components/payments/CheckoutForm.js [hunk 5]
```diff
@@ -92,13 +92,19 @@ class _CheckoutForm extends Component {
...
+                .then(() => {
+                    socket.emit('investment', {
+                        id: projectOwnerId,
+                        name: userName,
+                        projectName: projectTitle
+                    }, { token: localStorage.getItem('jwtToken') })
+                })
+                .then(() => activateToast(`Your payment was a success. Thank you for investing in ${projectTitle}!`, 'success'))
                 .then(() => history.push("/explore"))
                 .catch(() => activateToast('that was a fail', 'error'))
```
Comment (tsyaeger) [inline client/src/components/payments/CheckoutForm.js]: Evan was saying that we should be more professional with the messages.

File: server/sockets/events/Investments.js (hunk 2 above)
Comment (tsyaeger) [inline server/sockets/events/Investments.js]: not sure if you want to keep this

File: client/src/api/notifications.js (hunk 3 above)
Comment (DevDelvo) [reply client/src/api/notifications.js:3]: Okay Ive removed it for now and we'll see how it goes. No issue at the moment

**Review by evankp — APPROVED** [review]

**Pull request merged**